cmd/buildtools/utils.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/v62/github"
+	"github.com/sirupsen/logrus"
+)
+
+func GetLatestGitHubRelease(ctx context.Context, owner, repo string) (string, error) {
+	client := github.NewClient(nil)
+	release, _, err := client.Repositories.GetLatestRelease(ctx, owner, repo)
+	if err != nil {
+		return "", err
+	}
+	return release.GetName(), nil
+}
+
+func GetLatestGitHubTag(ctx context.Context, owner, repo string) (string, error) {
+	client := github.NewClient(nil)
+	tags, _, err := client.Repositories.ListTags(ctx, owner, repo, &github.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("unable to list tags: %w", err)
+	}
+	if len(tags) == 0 {
+		return "", fmt.Errorf("no tags found")
+	}
+	for _, tag := range tags {
+		if strings.Contains(tag.GetName(), "alpha") {
+			continue
+		}
+		if strings.Contains(tag.GetName(), "beta") {
+			continue
+		}
+		if strings.Contains(tag.GetName(), "rc") {
```

Comment:
what about instead checking that it is not a prerelease sem version (no "-")?
```suggestion
		// "-" indicates this is a pre-release version
		if strings.Contains(tag.GetName(), "-") {
```